docs/magento-layout-loader.md

```diff
@@ -0,0 +1,288 @@
+# magento-layout-loader
+
+_This is a very early implementation, and the API should be considered
+unstable._
+
+The `magento-layout-loader` is a [webpack
+loader](https://webpack.js.org/concepts/loaders/) that transforms
+[JSX](https://reactjs.org/docs/introducing-jsx.html) during compilation. It
+gives Magento modules/extensions the ability to inject or remove content blocks
+in a layout without modifying theme source files.
+
+## Terminology
+
+* **Container**: An HTML element that contains 0 or more ContainerChild
+  components. It acts as the target for the magento-loader-layout operations.
+* **ContainerChild**: Component exposed by
+  [Peregrine](https://github.com/magento-research/peregrine/). Responsible for
+  rendering content
+* **Operation**: An action that can be taken on a `Container` or
+  `ContainerChild`. Examples include `removeContainer`, `insertAfter`, etc.
+
+## `Container` Details
+
+A Container can be created by adding a `data-mid` prop to any DOM element
```

Comment:
I was wondering what does `mid` stand for ? Magento ID ?

If the `Container` component is used, I believe it could be simplified to `id` ? I don't know if would be easily feasible.

Reply:
`mid` is for Magento ID right now, but the naming isn't set in stone, and is up for bike-shedding.